homeassistant/components/hausbus/gateway.py

```diff
@@ -0,0 +1,185 @@
+"""Representation of a Haus-Bus gateway."""
+
+import asyncio
+from collections.abc import Callable, Coroutine
+from typing import Any, cast
+
+from pyhausbus.ABusFeature import ABusFeature
+from pyhausbus.BusDataMessage import BusDataMessage
+from pyhausbus.de.hausbus.homeassistant.proxy.Controller import Controller
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.Configuration import (
+    Configuration,
+)
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.ModuleId import ModuleId
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.RemoteObjects import (
+    RemoteObjects,
+)
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .device import HausbusDevice
+from .entity import HausbusEntity
+from .event_handler import IEventHandler
+from .light import (
+    Dimmer,
+    HausbusDimmerLight,
+    HausbusLedLight,
+    HausbusLight,
+    HausbusRGBDimmerLight,
+    Led,
+    RGBDimmer,
+)
+
+
+class HausbusGateway(IBusDataListener, IEventHandler):  # type: ignore[misc]
+    """Manages a single Haus-Bus gateway."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the system."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.bridge_id = "1"
+        self.devices: dict[str, HausbusDevice] = {}
+        self.channels: dict[str, dict[tuple[str, str], HausbusEntity]] = {}
+        self.home_server = HomeServer()
+        self.home_server.addBusEventListener(self)
+        self._new_channel_listeners: dict[
+            str, Callable[[HausbusEntity], Coroutine[Any, Any, None]]
+        ] = {}
+
+    def add_device(self, device_id: str, module: ModuleId) -> None:
+        """Add a new Haus-Bus Device to this gateways devices list."""
+        device = HausbusDevice(
+            self.bridge_id,
+            device_id,
+            module.getFirmwareId().getTemplateId()
+            + " "
+            + str(module.getMajorRelease())
+            + "."
+            + str(module.getMinorRelease()),
+            module.getName(),
+            module.getFirmwareId(),
+        )
+        if device_id not in self.devices:
+            self.devices[device_id] = device
+        if device_id not in self.channels:
+            self.channels[device_id] = {}
+
+    def get_device(self, object_id: ObjectId) -> HausbusDevice | None:
+        """Get the device referenced by ObjectId from the devices list."""
+        return self.devices.get(str(object_id.getDeviceId()), None)
```

Comment:
`dict.get` returns `None` if the key is missing, there's no need to pass `None`. Please change that here and in the other cases
```suggestion
        return self.devices.get(str(object_id.getDeviceId()))
```